apps/studio/pages/api/ai/sql/generate-v4.ts

```diff
@@ -102,6 +104,27 @@ async function handlePost(req: NextApiRequest, res: NextApiResponse) {
 
   const aiOptInLevel = getAiOptInLevel(selectedOrg?.opt_in_tags)
 
+  // Check enabled database extensions to potentially restrict data access
+  let effectiveAiOptInLevel = aiOptInLevel
+
+  try {
+    let headers = new Headers()
+    if (authorization) headers.set('Authorization', authorization)
+
+    const dbExtensions = await getDatabaseExtensions(
+      { projectRef, connectionString },
+      undefined,
+      headers
+    )
+
+    effectiveAiOptInLevel = checkNetworkExtensionsAndAdjustOptInLevel(
+      dbExtensions,
+      effectiveAiOptInLevel
+    )
+  } catch (error) {
+    console.error('Failed to fetch database extensions:', error)
```

Comment:
Would be safer to default downgrade opt-in in an error case, or let this bubble up and fail the entire request.